homeassistant/components/airtouch4/climate.py

```diff
@@ -0,0 +1,339 @@
+        self.async_write_ha_state()
+
+    async def async_turn_on(self):
+        """Turn on."""
+        _LOGGER.debug("Turning %s on", self.unique_id)
+        # in case ac is not on. Airtouch turns itself off if no groups are turned on
+        # (even if groups turned back on)
+        await self._airtouch.TurnAcOn(self._ac_number)
+
+    async def async_turn_off(self):
+        """Turn off."""
+        _LOGGER.debug("Turning %s off", self.unique_id)
+        await self._airtouch.TurnAcOff(self._ac_number)
+        self.async_write_ha_state()
+
+
+class AirtouchGroup(CoordinatorEntity, ClimateEntity):
+    """Representation of an AirTouch 4 group."""
+
+    _attr_supported_features = SUPPORT_TARGET_TEMPERATURE
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, coordinator, group_number, info):
+        """Initialize the climate device."""
+        super().__init__(coordinator)
+        self._group_number = group_number
+        self._airtouch = coordinator.airtouch
+        self._info = info
+        self._unit = self._airtouch.GetGroupByGroupNumber(self._group_number)
+
+    @callback
+    def _handle_coordinator_update(self):
+        self._unit = self._airtouch.GetGroupByGroupNumber(self._group_number)
+        return super()._handle_coordinator_update()
+
+    @property
+    def device_info(self):
+        """Return device info for this device."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Airtouch",
+            "model": "Airtouch 4",
+        }
+
+    @property
+    def unique_id(self):
+        """Return unique ID for this device."""
+        return self._group_number
+
+    @property
+    def min_temp(self):
+        """Return Minimum Temperature for AC of this group."""
+        return self._airtouch.acs[self._unit.BelongsToAc].MinSetpoint
+
+    @property
+    def max_temp(self):
+        """Return Max Temperature for AC of this group."""
+        return self._airtouch.acs[self._unit.BelongsToAc].MaxSetpoint
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._unit.GroupName
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._unit.Temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we are trying to reach."""
+        return self._unit.TargetSetpoint
+
+    @property
+    def hvac_mode(self):
+        """Return hvac target hvac state."""
+        # there are other power states that aren't 'on' but still count as on (eg. 'Turbo')
+        is_off = self._unit.PowerState == "Off"
+        if is_off:
+            return HVAC_MODE_OFF
+
+        return HVAC_MODE_FAN_ONLY
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available operation modes."""
+        return AT_GROUP_MODES
```

Comment:
This is static and should be set as a class attribute with `_attr_hvac_modes = AT_GROUP_MODES` on the `AirtouchGroup` class